integrations/wagmi-v2/src/index.ts

```diff
@@ -85,124 +68,95 @@ export function injectedWithSapphire(
 	} as InjectedParameters);
 }
 
+interface BaseConnector {
+	getProvider?(): Promise<unknown> | unknown;
+	name?: string;
+	id?: string;
+}
+
+type ConnectorFactoryReturn<C extends BaseConnector = BaseConnector> = C;
+
 /**
- * Creates a wagmi configuration with wrapped EIP-6963 Sapphire encryption layer.
- *
- * In case you are only using Sapphire chain:
- *
- * ```typescript
- * import { createSapphireConfig } from '@oasisprotocol/sapphire-wagmi-v2';
- *
- * export const config = createSapphireConfig({
- *   sapphireConfig: {
- *     replaceProviders: true,
- *   }
- *   ...
- * });
- * ```
- *
- * In case your dApp is multichain, the below example will create duplicate connectors for Sapphire:
+ * Wrap any Wagmi connector with the Sapphire encryption layer.
+ * Used to provide encrypted transactions and calldata to any connector type (WalletConnect, MetaMask, etc.).
  *
  * ```typescript
- * import { createSapphireConfig } from '@oasisprotocol/sapphire-wagmi-v2';
+ * import { wrapConnectorWithSapphire } from '@oasisprotocol/sapphire-wagmi-v2';
+ * import { walletConnect } from '@wagmi/connectors';
  *
- * export const config = createSapphireConfig({
- *   sapphireConfig: {
- *     replaceProviders: false,
- *     // Define which providers you want to wrap via RDNS
- *     wrappedProvidersFilter: (rdns) => ['io.metamask'].includes(rdns)
- *   }
+ * export const config = createConfig({
+ *   connectors: [
+ *     wrapConnectorWithSapphire(
+ *       walletConnect({ projectId: 'your-project-id' }),
```

Comment:
How do you wrap plain wagmi connector? e.g. If you don't want to use wallet connect / rainbowkit?

Reply:
With `wrapConnectorWithSapphire` utility function.